pkg/util/proxyutil/proxyutil.go

```diff
@@ -130,3 +132,32 @@ func ApplyForwardIDHeader(req *http.Request, user identity.Requester) {
 		req.Header.Set(IDHeaderName, token)
 	}
 }
+
+func ApplyTeamHeaders(req *http.Request, ds *datasources.DataSource, teams []int64) {
+	if ds.JsonData != nil {
+		for teamID, headers := range ds.JsonData.Get("teamHeaders").MustMap() {
+			id, err := strconv.ParseInt(teamID, 10, 64)
+			if err != nil {
+				// FIXME: logging here
+				continue
+			}
+			if !contains(teams, id) {
+				continue
+			}
+
+			v := headers.([]map[string]interface{})
+			for _, header := range v {
+				req.Header.Set(header["header"].(string), header["value"].(string))
```

Comment:
LBAC rules can be `OR`d by providing multiple headers. Using `(Header).Set` here causes multiple headers results in a single header (the last one).

I believe that a more sensible approach here would be to use `(Header).Set` for the initial header with a given name (so that existing/non-team headers get overwritten) and `(Header).Add` for subsequent cases.

Reply:
fixed in c21a9569a1